src/libs/actions/App.js

```diff
@@ -177,33 +167,39 @@ function fixAccountAndReloadData() {
  * pass it in as a parameter. withOnyx guarantees that the value has been read
  * from Onyx because it will not render the AuthScreens until that point.
  * @param {Object} session
+ * @param {string} currentPath
  */
-function setUpPoliciesAndNavigate(session) {
-    Linking.getInitialURL()
-        .then((url) => {
-            if (!url) {
-                return;
-            }
-            const path = new URL(url).pathname;
-            const params = new URLSearchParams(url);
-            const exitTo = params.get('exitTo');
-            const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(url, session.email);
-            const shouldCreateFreePolicy = !isLoggingInAsNewUser
-                        && Str.startsWith(path, Str.normalizeUrl(ROUTES.TRANSITION_FROM_OLD_DOT))
+function setUpPoliciesAndNavigate(session, currentPath) {
+    if (!session || !currentPath || !currentPath.includes('exitTo')) {
+        return;
+    }
+
+    let exitTo;
+    try {
+        const params = new URLSearchParams(currentPath);
+        exitTo = params.get('exitTo');
+    } catch (error) {
+        // URLSearchParams is unsupported on iOS so we catch th error and
+        // silence it here since this is primarily a Web flow
+        return;
+    }
+
+    const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(currentPath, session.email);
+    const shouldCreateFreePolicy = !isLoggingInAsNewUser
+                        && Str.startsWith(currentPath, Str.normalizeUrl(ROUTES.TRANSITION_FROM_OLD_DOT))
                         && exitTo === ROUTES.WORKSPACE_NEW;
-            if (shouldCreateFreePolicy) {
-                Policy.createAndGetPolicyList();
-                return;
-            }
-            if (!isLoggingInAsNewUser && exitTo) {
-                Navigation.isNavigationReady()
-                    .then(() => {
-                        // We must call dismissModal() to remove the /transition route from history
-                        Navigation.dismissModal();
-                        Navigation.navigate(exitTo);
-                    });
-            }
-        });
+    if (shouldCreateFreePolicy) {
+        Policy.createAndGetPolicyList();
+        return;
+    }
+    if (!isLoggingInAsNewUser && exitTo) {
+        Navigation.isNavigationReady()
```

Comment:
NAB: We can probably get rid of this now that it only runs once the navigation is ready. You'll have to test. Could be a follow up PR.

Reply:
Agree with removing this if we can.